doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,456 @@
+
+    dev = qml.device("default.qubit")
+
+    @qml.qnode(dev)
+    def circuit(phi, theta):
+        qml.RX(phi, wires=0)
+        m_0 = qml.measure(wires=0)
+        qml.RY(theta, wires=1)
+        m_1 = qml.measure(wires=1)
+        return qml.sample([m_0, m_1])
+
+Executing this ``QNode``:
+
+.. code-block:: pycon
+
+    >>> circuit(1.23, 4.56, shots=5)
+    array([[0, 1],
+           [1, 1],
+           [0, 1],
+           [0, 0],
+           [1, 1]])
+
+Collecting statistics for sequences of mid-circuit measurements is supported with
+:func:`~.counts`, :func:`~.probs`, and :func:`~.sample`.
+
+.. warning::
+
+    When collecting statistics for a sequence of mid-circuit measurements, the
+    sequence may not contain arithmetic expressions.
+
+.. _simulation_techniques:
+
+Simulation techniques
+---------------------
+
+PennyLane currently offers three methods to simulate mid-circuit measurements
+on classical computers: the deferred measurements principle, dynamic one-shot
+sampling, and a tree-traversal approach. These methods differ in their memory requirements
+and computational cost, as well as their compatibility with other features such as
+shots and differentiation methods.
+While the requirements depend on details of the simulation, the expected
+scalings  with respect to the number of mid-circuit measurements (and shots) are
+
+.. role:: gr
+.. role:: or
+.. role:: rd
+
+.. raw:: html
+
+   <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
+   <script>
+     $(document).ready(function() {
+       $('.gr').parent().parent().addClass('gr-parent');
+       $('.or').parent().parent().addClass('or-parent');
+       $('.rd').parent().parent().addClass('rd-parent');
+     });
+   </script>
+   <style>
+       .gr-parent {background-color:#e1eba8}
+       .or-parent {background-color:#ffe096}
+       .rd-parent {background-color:#ffb3b3}
+       .tb { border-collapse: collapse; }
+       .tb th, .tb td { padding: 1px; border: solid 1px black; vertical-align: middle; column-width: auto; }
+   </style>
+
+.. rst-class:: tb
+
++--------------------------+-------------------------------------------+-----------------------------------------------------------+-------------------------------------------+----------------------------------+
+| **Simulation technique** | **Memory**                                | **Time**                                                  | **Differentiation support**               | **Supports shots/analytic mode** |
++==========================+===========================================+===========================================================+===========================================+==================================+
+| Deferred measurements    | :rd:`\ ` :math:`\mathcal{O}(2^{n_{MCM}})` | :rd:`\ ` :math:`\mathcal{O}(2^{n_{MCM}})`                 | :gr:`\ ` yes                              | :gr:`\ ` yes / yes               |
```

Comment:
Here we say that differentiation is supported with `defer_measurements`, but this is only _always_ true for backprop, and adjoint/parameter shift are only supported without postselection since `Projector`'s are not unitary.

Reply:
Took the liberty to squash this information into a footnote like for the other sim. methods.